packages/hash/api/src/graphql/resolvers/ontology/data-type.ts

```diff
@@ -1,51 +1,69 @@
 import { ApolloError } from "apollo-server-express";
 import { AxiosError } from "axios";
 
-import {
-  PersistedDataType,
-  QueryGetDataTypeArgs,
-  ResolverFn,
-} from "../../apiTypes.gen";
+import { Subgraph, QueryGetDataTypeArgs, ResolverFn } from "../../apiTypes.gen";
 import { GraphQLContext, LoggedInGraphQLContext } from "../../context";
-import { DataTypeModel } from "../../../model";
-import { mapDataTypeModelToGQL } from "./model-mapping";
 
 export const getAllLatestDataTypes: ResolverFn<
-  Promise<PersistedDataType[]>,
+  Promise<Subgraph>,
   {},
   LoggedInGraphQLContext,
   {}
 > = async (_, __, { dataSources }) => {
   const { graphApi } = dataSources;
 
-  const allLatestDataTypeModels = await DataTypeModel.getAllLatest(
-    graphApi,
-  ).catch((err: AxiosError) => {
-    throw new ApolloError(
-      `Unable to retrieve all latest data types. ${err.response?.data}`,
-      "GET_ALL_ERROR",
-    );
-  });
+  const { data: dataTypeSubgraph } = await graphApi
+    .getDataTypesByQuery({
+      query: { eq: [{ path: ["version"] }, { literal: "latest" }] },
+      /** todo - make these configurable once non-primitive data types are a thing https://app.asana.com/0/1200211978612931/1202464168422955/f */
+      queryResolveDepths: {
+        dataTypeResolveDepth: 0,
+        propertyTypeResolveDepth: 0,
+        linkTypeResolveDepth: 0,
+        entityTypeResolveDepth: 0,
+        entityResolveDepth: 0,
+        linkResolveDepth: 0,
+      },
+    })
+    .catch((err: AxiosError) => {
+      throw new ApolloError(
+        `Unable to retrieve all latest data types: ${err.response?.data}`,
+        "GET_ALL_ERROR",
+      );
+    });
 
-  return allLatestDataTypeModels.map(mapDataTypeModelToGQL);
+  return dataTypeSubgraph;
 };
 
 export const getDataType: ResolverFn<
-  Promise<PersistedDataType>,
+  Promise<Subgraph>,
   {},
   GraphQLContext,
   QueryGetDataTypeArgs
 > = async (_, { dataTypeId }, { dataSources }) => {
   const { graphApi } = dataSources;
 
-  const dataTypeModel = await DataTypeModel.get(graphApi, {
-    dataTypeId,
-  }).catch((err: AxiosError) => {
-    throw new ApolloError(
-      `Unable to retrieve data type. ${err.response?.data}`,
-      "GET_ERROR",
-    );
-  });
+  const { data: dataTypeSubgraph } = await graphApi
+    .getDataTypesByQuery({
+      query: {
+        eq: [{ path: ["versionedUri"] }, { literal: dataTypeId }],
+      },
+      /** todo - make these configurable once non-primitive data types are a thing https://app.asana.com/0/1200211978612931/1202464168422955/f */
+      queryResolveDepths: {
+        dataTypeResolveDepth: 0,
+        propertyTypeResolveDepth: 0,
+        linkTypeResolveDepth: 0,
+        entityTypeResolveDepth: 0,
+        entityResolveDepth: 0,
+        linkResolveDepth: 0,
+      },
+    })
+    .catch((err: AxiosError) => {
+      throw new ApolloError(
+        `Unable to retrieve data type [${dataTypeId}]: ${err.response?.data}`,
+        "GET_ALL_ERROR",
```

Comment:
This wouldn't be a `GET_ALL_ERROR`, so perhaps switching back to
```suggestion
        "GET_ERROR",
```